src/OpenFOAM/meshes/primitiveMesh/PrimitivePatch/PrimitivePatch.H

```diff
@@ -307,7 +307,10 @@ public:
             return points_;
         }
 
-
+        const gpuField<PointType>& getPoints() const
+        {
+            return gpuField<PointType>(points_);
```

Comment:
This is an error - you are returning a reference to a temporary variable. The compiler should have reported a warning here.